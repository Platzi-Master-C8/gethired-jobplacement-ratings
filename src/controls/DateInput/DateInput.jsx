import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import TextField from '@mui/material/TextField';
import DateAdapter from '@mui/lab/AdapterMoment';
import LocalizationProvider from '@mui/lab/LocalizationProvider';
import TimePicker from '@mui/lab/TimePicker';
import DateTimePicker from '@mui/lab/DateTimePicker';
import DatePicker from '@mui/lab/DatePicker';

const DateInput = ({ value, onChange, label, name, disabled, withTime, justTime, ...props }) => {
    const handleChange = (val) => {
        const e = { target: { value: moment(val).format('YYYY-MM-DD'), name } };
        onChange(e);
    };

    if (justTime)
        return (
            <LocalizationProvider dateAdapter={DateAdapter}>
                <TimePicker
                    disableFuture
                    id={name}
                    name={name}
                    label={label}
                    value={value}
                    disabled={disabled}
                    onChange={handleChange}
                    sx={{ width: '100%' }}
                    renderInput={(params) => <TextField {...params} />}
                    {...props}
                />
            </LocalizationProvider>
        );

    return (
        <LocalizationProvider dateAdapter={DateAdapter}>
            {withTime ? (
                <DateTimePicker
                    disableFuture
                    id={name}
                    name={name}
                    label={label}
                    value={value}
                    disabled={disabled}
                    onChange={handleChange}
                    sx={{ width: '100%' }}
                    renderInput={(params) => <TextField {...params} />}
                    {...props}
                />
            ) : (
                <DatePicker
                    disableFuture
                    id={name}
                    name={name}
                    label={label}
                    value={value}
                    disabled={disabled}
                    onChange={handleChange}
                    sx={{ width: '100%' }}
                    renderInput={(params) => <TextField {...params} />}
                    {...props}
                />
            )}
        </LocalizationProvider>
    );
};

DateInput.propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    label: PropTypes.string,
    disabled: PropTypes.bool,
    withTime: PropTypes.bool,
    justTime: PropTypes.bool,
};

DateInput.defaultProps = {
    label: '',
    disabled: false,
    withTime: false,
    justTime: false,
};

export default DateInput;
