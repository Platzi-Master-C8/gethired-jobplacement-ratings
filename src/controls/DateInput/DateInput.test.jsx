import React from 'react';
import moment from 'moment';

import * as driver from '@testing-library/react';
import DateInput from './DateInput';

const defaultProps = {
    value: moment().subtract(1, 'M').format('YYYY-MM-DD'),
    onChange: () => {},
    label: 'Test DateInput',
    name: 'Test DateInput',
    disabled: false,
    withTime: false,
    justTime: false,
};

const renderComponent = (props = {}) => driver.render(<DateInput {...defaultProps} {...props} />);

describe('Component <DateInput/>', () => {
    test('renders DateInput', () => {
        const { asFragment } = renderComponent({});
        expect(asFragment()).toMatchSnapshot();
    });
});
