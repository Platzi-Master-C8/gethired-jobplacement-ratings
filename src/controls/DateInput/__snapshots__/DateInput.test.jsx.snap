// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <DateInput/> renders DateInput 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="mui-1"
      id="mui-1-label"
    >
      Test DateInput
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        aria-label="Choose date, selected date is Dec 28, 2021"
        aria-readonly="true"
        class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="mui-1"
        readonly=""
        type="text"
        value="12/28/2021"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1z7n62"
        >
          <span>
            Test DateInput
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
